modules/zapier/class-module-zapier.php

```diff
@@ -79,13 +79,32 @@ public function pull_the_trigger( array $data, $hook_url ) {
              *
              * @since    1.1.0
              */
-            $result = wp_remote_post( $hook_url, apply_filters( 'ctz_post_request_args', $args ) );
-
-            // If result is a WP Error, throw a Exception woth the message.
-            if ( is_wp_error( $result ) ) {
-                throw new Exception( $result->get_error_message() );
-            }
+            $response = wp_remote_post( $hook_url, apply_filters( 'ctz_post_request_args', $args ) );
+			$body = wp_remote_retrieve_body( $response );
+			$data = json_decode( $body, true );
+			
+			$statusCode = $data['status'];
+			$errorCode = $data['error_code'];
+			$errorMessage = $data['message'];
 
+			if ($statusCode >= 400) {
+				$errorMsg = 'This submission is not valid';
+				
+				if ($statusCode == 400) {
+					if (strpos($errorMessage, 'and owns goods') !== false || strpos($errorMessage, 'Email already signed up') !== false) {
+						$errorMsg = "It seems like you are already a Bookful subscriber.<br>Please contact support@inceptionxr.com to learn how you can enjoy this exclusive offer.";
+					} 
+// 					elseif ($errorCode == 7002) {
```

Comment:
commented do nothing
do not see reason to keep it

Reply:
I think that we would need them soon, as those case are not handled by product request at the moment, but it's better to leave it there as a reference